src/include/btree.i

```diff
@@ -1435,7 +1452,7 @@ __wt_page_release(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
      * evict it.
      */
     page = ref->page;
-    if (WT_READGEN_EVICT_SOON(page->read_gen) && btree->evict_disabled == 0 &&
```

Comment:
Can the `WT_READGEN_EVICT_SOON` macro go away?  It seems confusing and error prone to keep two different tests...

Reply:
It's used in at least 5 other places so I'll have to go through each scenario and determine if that's a valid use case for the new function.

Reply:
There are scenarios where we only have the readgen available, i.e. no page reference and use the macro. In which case the function wouldn't suit, I could change it to handle that case too but I think it would be best to keep both?